sunpy/map/sources/suit.py

```diff
@@ -0,0 +1,121 @@
+"""
+SUIT Map sub class definitions
+
+"""
+from sunpy.map.sources.source_type import source_stretch
+import numpy as np
+import astropy.units as u
+from astropy.coordinates import CartesianRepresentation, HeliocentricMeanEcliptic
+from sunpy.map.mapbase import GenericMap
+from matplotlib import colors
+from astropy.visualization import ImageNormalize
+from astropy.visualization import AsinhStretch
+
+__all__ = ['SUITMap']
+__author__ = ['Rahul Gopalakrishnan']
+__email__ = ['rahulg.astro@gmail.com']
+
+class SUITMap(GenericMap):
+    """
+    The Solar Ultraviolet Imaging Telescope (SUIT) is one of the remote sensing payloads on board the Aditya-L1 mission of the Indian Space Research Organization (ISRO)
+    that was launched on September 02, 2023. SUIT is designed to provide near-simultaneous full-disk and region-of-interest images of the Sun at various heights,
+    slicing through the photosphere and chromosphere, employing an array of 11 scientifically calibrated filters (3 broad-band & 8 narrow-band) strategically
+    positioned within the wavelength range of 200 to 400 nanometers. Located at the first Lagrange point, SUIT observes the Sun 24x7, without any interruption.
+
+    References:
+    -----------
+    * `Home Page: <https://suit.iucaa.in/>`_
+    * `Mission Page: <https://suit.iucaa.in/about_SUIT>`_
+    * `Publications Page: <https://suit.iucaa.in/node/5>`_
+    """
+
+    def __init__(self, data, header, **kwargs):
+        super().__init__(data, header, **kwargs)
+        self._nickname = self.detector
+        self.filter = header.get('FTR_NAME').strip()
+        self.wavelnth = str(header.get('WAVELNTH','NA')).strip()
+        self.plot_settings['cmap'] = f"suit_{self.filter.lower()}"
+        self.plot_settings['title'] = f"SUIT {self.filter}:{self.wavelnth} A - {self.reference_date}"
+        norm = self.suit_norm()
+
+        if self.filter in norm:
+            self.plot_settings["norm"] = norm[self.filter]
+        else:
+            self.plot_settings["norm"] = ImageNormalize(
+                stretch=source_stretch(self.meta, AsinhStretch(0.01)), clip=False
+            )
+
+    def suit_norm(self):
```

Comment:
This section subtly breaks one of the conventions we have for map subclasses, which is that the subclasses shouldn't add any API which isn't in `GenericMap`, so in this case it means you shouldn't be adding public attributes to `self` here.

```suggestion
        filter = header.get('FTR_NAME').strip()
        wavelnth = str(header.get('WAVELNTH', 'NA')).strip()
        self.plot_settings['cmap'] = f"suit_{filter.lower()}"
        self.plot_settings['title'] = f"SUIT {filter}:{wavelnth} A - {self.reference_date}"
        norm = self._suit_norm()

        if self.filter in norm:
            self.plot_settings["norm"] = norm[filter]
        else:
            self.plot_settings["norm"] = ImageNormalize(
                stretch=source_stretch(self.meta, AsinhStretch(0.01)), clip=False
            )

    def _suit_norm(self):
```